src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1363,6 +1351,8 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
    std::string encoding = projects::projectContext().hasProject() ?
                           projects::projectContext().defaultEncoding() :
                           prefs::userPrefs().defaultEncoding();
+   if (encoding.empty())
+      encoding = "UTF-8";
```

Comment:
Is this assumption valid on Windows?

Reply:
Yes, the only places we use the encoding are with `r::util::iconvstr` and the first thing that function does is change empty encodings to UTF-8. https://github.com/rstudio/rstudio/blob/master/src/cpp/r/RUtil.cpp#L143-L160

Reply:
I don't think so -- on Windows, outside of an RStudio project, files will be saved in the system encoding, which will not be UTF-8. To that end, I think we need to be able to call R's `iconv()` machinery and tell it to conver from the native encoding to UTF-8. This is normally done by passing an empty string to `Riconv_open()`(https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Re_002dencoding).

We do a strange here to check if the "effective" encodings are the same, to basically avoid unnecessary conversions:

https://github.com/rstudio/rstudio/blob/db6105f69c895b4c96d426ed95ef77bde14ff574/src/cpp/r/RUtil.cpp#L149-L160

but this check is also wrong (at least, on Windows; but also in theory on other platforms where a non-UTF8 locale has been set).

In short, I think:

1. It's okay to leave the encoding here as the empty string;
2. The shortcut in `iconvstr()` should avoid assuming that "empty string" -> "UTF-8".

Reply:
I updated the code with Kevin's recommendation.